### PR TITLE
Fix passing kwargs to TFBertTokenizer

### PR DESCRIPTION
**Pull request by balvisio**
# What does this PR do?

This PR fixes passing `kwargs` when creating a `TFBertTokenizer.from_pretrained()`. Currently, a `kwarg` is passed the following error is raised:

```
>>> from transformers import TFBertTokenizer
>>> tokenizer = TFBertTokenizer.from_pretrained("distilbert-base-cased", do_lower_case=False)

TypeError: transformers.models.bert.tokenization_bert_tf.TFBertTokenizer() got multiple values for keyword argument 'vocab_list' 
```

By popping the arguments from `kwargs` we avoid the ambiguity.

@ArthurZucker  You might be interested in this PR. Not clear to me which `kwargs` should actually be allowed when calling `from_pretrained()`. For example, `do_lower_case` makes sense but not sure if `vocab_list` or `cls_token_id` should be even allowed?

Thanks!

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/bert/tokenization_bert_tf.py [hunk 1]
```diff
@@ -114,6 +114,11 @@ def from_tokenizer(cls, tokenizer: "PreTrainedTokenizerBase", **kwargs):  # noqa
         tf_tokenizer = TFBertTokenizer.from_tokenizer(tokenizer)
         ```
         """
+        kwargs.pop("do_lower_case", None)
+        kwargs.pop("cls_token_id", None)
+        kwargs.pop("sep_token_id", None)
+        kwargs.pop("pad_token_id", None)
```
Comment (ArthurZucker) [inline src/transformers/models/bert/tokenization_bert_tf.py:120]: I think we can do something like: 
```suggestion
        do_lower_case = kwargs.pop("do_lower_case", None)
        do_lower_case = tokenizer.do_lower_case if do_lower_case is None else do_lower_case
```
for these kwargs. Removing them completely is prone to errors, and people using them but not getting what they want. WDYT 😉

**Comment by sgugger** [comment]
Also cc @Rocketknight1

File: src/transformers/models/bert/tokenization_bert_tf.py (hunk 1 above)
Comment (balvisio) [reply src/transformers/models/bert/tokenization_bert_tf.py]: Yeah, I agree with your suggestions. I was considering the `from_pretained()` path but this solution is correct also when `from_tokenizer()` is called directly by the user.

**Review by ArthurZucker — APPROVED** [review]
LGTM let's wait for the other reviewer

**Review by Rocketknight1 — APPROVED** [review]
This looks like a clean PR to me. I definitely overlooked this when I wrote the original, so thanks for fixing it!

**Review by sgugger — APPROVED** [review]
Thanks for your contribuyion!

**Pull request merged**